tests/foreman/ui/test_errata.py

```diff
@@ -247,6 +254,75 @@ def test_end_to_end(session, module_repos_col, vm):
         assert result['result'] == 'success'
 
 
+@pytest.mark.tier2
+@pytest.mark.skipif((not settings.repos_hosting_url), reason='Missing repos_hosting_url')
+def test_content_host_errata_page_pagination(session, module_org, module_lce):
+    """
+    # Test per-page pagination for BZ#1662254
+    # Test apply by REX using Select All for BZ#1846670
+
+    :Steps:
+        1. Install more than 20 packages that need errata
+        2. View Content Host's Errata page
+        3. Assert total_pages > 1
+        4. Change per-page setting to 100
+        5. Assert table has more than 20 errata
+        6. Use the selection box on the left to select all on the page.
+            The following is displayed at the top of the table:
+            All 20 items on this page are selected. Select all YYY.
+
+        7. Click the "Select all" text and assert more than 20 results are selected.
+        8. Click the drop down arrow to the right of "Apply Selected", and select
+            "via remote execution - customize first "
+        9. Assert all the errata is in the "errata" field.
+        10. Click Submit
+        11. Assert Errata are applied as expected.
+
+    :BZ: 1662254, 184667
+    """
+    org = module_org
+    lce = entities.LifecycleEnvironment(organization=org).create()
+    pkgs = " ".join(FAKE_3_YUM_OUTDATED_PACKAGES)
+    repos_collection = RepositoryCollection(
+        distro=DISTRO_RHEL7,
+        repositories=[
+            SatelliteToolsRepository(),
+            # As Satellite Tools may be added as custom repo and to have a "Fully entitled" host,
+            # force the host to consume an RH product with adding a cdn repo.
+            # RHELAnsibleEngineRepository(cdn=True),
+            YumRepository(url=FAKE_3_YUM_REPO),
+        ],
+    )
+    repos_collection.setup_content(org.id, lce.id, upload_manifest=True)
+    with VirtualMachine(distro=repos_collection.distro) as client:
+        # add_remote_execution_ssh_key for REX
+        add_remote_execution_ssh_key(client.ip_addr)
+        # Add repo and install packages that need errata
+        repos_collection.setup_virtual_machine(client)
+        assert _install_client_package(client, pkgs)
+        with session:
+            # Go to CHost's Errata page and read the page's pagination widgets
+            page_values = session.contenthost.read(client.hostname, widget_names=['errata.pagination'])
+            # assert total_pages > 1 with default page settings
+            assert int(page_values['errata']['pagination']['pages']) > 1
+            # Change per-page setting to 100
+            ch = session.contenthost
+            view = ch.navigate_to(ch, 'Edit', entity_name=client.hostname)
+            per_page_value = view.errata.pagination.per_page
+            assert int(per_page_value.read()) == 20
+            assert per_page_value.fill('100')
```

Comment:
Usually, we try to avoid using Airgun internals (like navigation and specific widgets) inside Robottelo. Sometimes there's no other way, but here it looks like `update()` and `read()` should suffice?

```python
page_values = session.contenthost.read(client.hostname, widget_names=['errata.pagination'])
assert int(page_values['errata']['pagination']['per_page']) == 20
session.contenthost.update({'errata.pagination.per_page': '100'})
```

Reply:
This does not work:
session.contenthost.update({'errata.pagination.per_page': '100'})
TypeError: update() missing 1 required positional argument: 'values'

Reply:
@swadeley You should be able to resolve that by passing the dict as `values` kwarg.

Reply:
Hi, yes figured out it probably should be :
 `session.contenthost.update(client.hostname, {'errata.pagination.per_page': '100'})`
but then that erros:
AttributeError: Object <<class 'airgun.widgets.Pagination'>> has no widget name "per_page"

Reply:
@swadeley that's probably because widgets returned by decorators are not properly included within `Widget.widget_names`. I wouldn't lose sleep over it, if `update()` is not good enough in this case, then it's not good enough in this case.

Reply:
> @swadeley please fix Travis/GitHub Actions failures and I think we are good to merge.

@mirekdlugosz Change made (it was the dreaded `Reorder python imports` failure).

> I think your test results already include [SatelliteQE/fedorapeople-repos#7](https://github.com/SatelliteQE/fedorapeople-repos/pull/7) , as without this change this test fails during setup?

Correct, all test runs have always included the use of the new yum3 repo

Thank you